allen-cell-animated/nucmorph-colorizer
# refactor: Header component

Problem
=======
Refactors the header into a reusable series of components as part of #250.

*Estimated review size: small, <15 minutes. (Recommend turning off whitespace)*

Solution
========
- Moves the header bar into its own `Header.tsx` component file, which pages can reuse.
  - We can't mount the header as part of the root component, because many of the buttons that go in the header (export, copy, etc.) are context-specific and require access to page state.
  - This component will eventually be used by the landing page.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link and confirm that the inputs and interactions in the header work: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-257/
2. Open the error page. The header should also appear: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-257/#/bad

Screenshots (optional):
-----------------------
![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/38f118ee-a325-4d5c-b316-900bfaaa05cc)
